example/simple_testing/kotlin/src/main/kotlin/dev/diplomattest/simple_testing/Main.kt

```diff
@@ -0,0 +1,54 @@
+package dev.diplomattest.simple_testing;
+
+import com.sun.jna.Callback
+import com.sun.jna.Library
+import com.sun.jna.Native
+import com.sun.jna.Pointer
+
+
+interface DiplomatCallback_Lib: Library {
+    // general use
+    fun diplomat_callback_create_for_jvm__callback(callback_wrap: Callback): Pointer
+
+    // specific to this callback type
+    interface DiplomatCallbackI32ToI32: Callback {
+        fun invoke(input: Int): Int
+    }
+    // specific to this callback
+    fun DiplomatCallbackI32ToI32_test_rust_fn_test_call(diplomatCallback: Pointer): Int
+}
+
+class DiplomatCallback internal constructor (
+    internal val handle: Pointer,
+    // These ensure that anything that is borrowed is kept alive and not cleaned
+    // up by the garbage collector.
+    // Note: ^ this comment was copied from other diplomat code
+    internal val selfEdges: List<Any>
+)  {
+    companion object {
+        internal val libClass: Class<DiplomatCallback_Lib> = DiplomatCallback_Lib::class.java
+        internal val lib: DiplomatCallback_Lib = Native.load("somelib", libClass)
+
+        // create the JNA Callback object
+        internal val callback: DiplomatCallback_Lib.DiplomatCallbackI32ToI32 = object : DiplomatCallback_Lib.DiplomatCallbackI32ToI32{
+            override fun invoke(input: Int): Int {
+                return input*2;
+            }
+        }
```

Comment:
I left a similar comment on the design doc, but would the use case for providing callbacks be that as a library user I can create the callback in my backend language (Kotlin in this case) and then create a `DiplomatCallback` instance that I can pass around to other FFI methods/functions? So I would expect you'd want generated code to create something like this (for each possible signature in the rust library we're running the diplomat tool on)
```kotlin
fun fromFunction(cb: (Int) -> Int): DiplomatCallbackIntToInt {
  val callback: DiplomatCallback_Lib.DiplomatCallbackI32ToI32 = object :  DiplomatCallback_Lib.DiplomatCallbackI32ToI32 {
      override fun invoke(input: Int): Int {
          return cb(input);
      }
    }
    val callback_pointer = lib.diplomat_callback_create_for_jvm__callback(callback);
    return DiplomatCallback(callback_pointer, {callback})
  }
}
```

Then any kotlin method generated from a rust method expecting a callback would then expect a `DiplomatCallback`
```kotlin
interface MyClassLib: Library {
  // general use
  fun MyClass_test_rust_fn_with_callback(cb: Pointer): Int;
}

class MyClass {
  companion object {
    internal val lib: ....
    fun testRustFnWithCallback(cbWrap: DiplomatCallbackIntToInt): Int {
      return lib.MyClass_test_rust_fn_with_callback(cbWrap.pointer)
    }
  }
}
```

Which would be generated from this.
```rust
#[diplomat::bridge]
mod ffi {
  
  pub struct MyClass;

  impl MyClass {
    pub fn test_rust_fn_with_callback(cb: impl Fn(i32) -> i32) -> i32 {
      cb(10)
    }
  }
}
```

Then someone comes along and wants to use the kotlin library generated from rust code. They would want to provide their own callback, like so:
```kotlin
object Main {
  @JvmStatic
  fun main(args: Array<String>) {
    val callBack = DiplomatCallback.fromFun( (input) -> input * 2))
    MyClass.testRustFnWithCallback(callBack)
  }
}
```
Or would we rather expect that the generated method takes a `fun` directly:
```kotlin
    fun testRustFnWithCallback(cbWrap: (Int) -> Int): Int {
      val cbWrapDiplomatCallback = DiplomatCallback.fromFun(cbWrap))
      return lib.MyClass_test_rust_fn_with_callback(cbWrapDiplomatCallback.pointer)
    }
```
and someone can come along and do
```kotlin
object Main {
  @JvmStatic
  fun main(args: Array<String>) {
    MyClass.testRustFnWithCallback({input -> 2 * input})
  }
}
```